### ISSUE TITLE
Support Groovy language

### ISSUE THREAD

**Issue by JLLeitschuh** [autonomousapps/dependency-analysis-android-gradle-plugin#364]
**Build scan link**

https://ge.gradle.org/s/qon5adsq33wtm

**Plugin version**

`0.70.0`

**Gradle version**

```
------------------------------------------------------------
Gradle 7.0-20210206125447+0000
------------------------------------------------------------

Build time:   2021-02-06 12:54:47 UTC
Revision:     99b7e817a96482acfc9e1b898dcb2919cd3938e8

Kotlin:       1.4.30
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.8 (AdoptOpenJDK 11.0.8+10)
OS:           Mac OS X 10.15.7 x86_64

```

**Android Gradle Plugin (AGP) version**
N/A

**Describe the bug**

There are false unneeded dependency reports when dependencies are needed in Groovy.

**To Reproduce**
1. Checkout `gradle/gradle` and execute `buildHealth`.
2. Observe the following report output:
   ```
   Advice for project :internal-testing
   Unused dependencies which should be removed:

   - implementation("org.slf4j:slf4j-api:1.7.28")
   - implementation("org.testcontainers:spock:1.12.5")
   - implementation("org.jsoup:jsoup:1.11.3")
   - implementation("org.ow2.asm:asm-tree:7.3.1")
   - implementation("org.ow2.asm:asm:7.3.1")
   ```
3. Follow advice.
4. Run `./gradlew :internal-testing:sanityCheck`
5. Observe Groovy compiler errors.

**Expected behavior**

Dependencies required by Groovy compiler should be accurately captured.

**Additional context**

Thanks for this awesome plugin!

**Comment by autonomousapps** [comment]
Thanks for the report! As discussed in slack, we maybe be able to make good progress by treating a groovy project like a java project, but we could run into issues if the groovy code doesn't reference an external class in a static fashion, but rather via reflection. If, for example, the groovy bytecode uses `Class.forName("com.foo.Bar")`, then we cannot detect the class `com/foo/Bar` directly in the consuming bytecode. This should not be much of a problem (I think) for classes with `@CompileStatic`.

**Comment by autonomousapps** [comment]
Looking at this, I don't see why it wouldn't work. So long as you also apply either the `java-library` or `application` plugins, then the fact that the code is written in Groovy shouldn't matter _except_ for potential issues with Groovy dynamism. Since this issue was reported over a year ago and this plugin has been rewritten practically from scratch in the meantime, I'm going to opportunistically close this issue. Please re-open if there are reproducible issues.

**Comment by autonomousapps** [comment]
Hm, upon reflection, there is still missing functionality. Need to analyze Groovy source to really be able to say this plugin can handle Groovy projects.

**Comment by autonomousapps** [comment]
This will be released as 1.4.0.
